bootstrap/result.go

```diff
@@ -65,6 +65,9 @@ func (sr *shardResult) AddSeries(id string, rawSeries m3db.DatabaseSeriesBlocks)
 
 // AddResult adds a shard result.
 func (sr *shardResult) AddResult(other m3db.ShardResult) {
+	if other == nil {
+		return
+	}
```

Comment:
Ok, sounds good.  To be super explicit we could've named this `AddAllSeriesFromResult(...)` and then it would be kind of clear that if the result is actually `nil` that it should be a noop (as you're adding a range of data, not another element of data). 

I'm not fussed though, if you think this is idiomatic then let's keep it.